### ISSUE TITLE
[Bug] Picker control does not announce number of items or position of selected item

### ISSUE THREAD

**Issue by samhouts** [xamarin/Xamarin.Forms#15396]
### Description

Expected
If items are grouped in a semantic list, the number of elements in that list is announced to most screen readers as well as your current selection in the total list. Additionally, you can use the role of list and listitem to create a list of objects as well. Note that most cellphone screen readers do not announce position information.

User Experience
Knowing how many items are in a list is extremely useful to assistive technology users as they can know if they should attempt to skip the list, navigate to the end, help them recall placement of certain items, or orient themselves within the list.

### Steps to Reproduce

1. Add picker to screen with n items.
2. Launch app with TalkBack enabled.
3. Activate picker. Move through items.

### Expected Behavior

TalkBack announces "item x of n" when item is highlighted.

### Actual Behavior

Doesn't do that ^

### Basic Information

- Version with issue: 5.0
- Last known good version: n/a
- Platform Target Frameworks: 
  - Android

**Comment by rachelkang** [comment]
This might be related to our findings in https://github.com/dotnet/maui/issues/6498 where we learned that TalkBack expected behavior has changed
